src/prog_algs/state_estimators/kalman_filter.py

```diff
@@ -66,8 +67,17 @@ def __init__(self, model, x0, measurement_eqn = None, **kwargs):
 
         self.filter = kalman.KalmanFilter(num_states, num_measurements, num_inputs)
 
-        self.filter.x = np.array([[x0[key]] for key in model.states])
-        self.filter.P = self.parameters['Q'] / 10
+        if isinstance(x0, dict):
+            warn("Warning: Use UncertainData type if estimating filtering with uncertain data.")
+            self.filter.x = np.array([[x0[key]] for key in model.states])
+            self.filter.P = self.parameters['Q'] / 10
+        elif isinstance(x0, UncertainData):
+            x_mean = x0.mean
+            self.filter.x = np.array([[x_mean[key]] for key in model.states])
+            self.filter.P = x0.cov
```

Comment:
We might have an ordering issue here. is x_mean.keys() is not the same order as model.states, then x0.cov will be in the wrong order. 

For example:
MultivariateNormalDist(['def', 'abc'], [2, 1], [[1, 0], [0, 2]])
model.states = ['abc', 'def']

then self.filter.x == [[1], [2]] and self.filter.P = [[1, 0], [0, 2]]. I'll add a test for this.